### PR TITLE
helm3 only support for all charts

### PR DESCRIPTION
**Pull request by hasancelik**
After this PR;
```
$ helm2 version
Client: &version.Version{SemVer:"v2.16.7", GitCommit:"5f2584fd3d35552c4af26036f0c464191287986b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.16.7", GitCommit:"5f2584fd3d35552c4af26036f0c464191287986b", GitTreeState:"clean"}

$ helm2 install --name hz-v2 hazelcast-4.0.0.tgz
Error: apiVersion 'v2' is not valid. The value must be "v1"
```

- [ ] README updates by removing helm2 references and adding warning notes about Helm3 support

### CODE REVIEW

**Comment by leszko** [comment]
@hasancelik @mesutcelik , so what plan to not support Helm 2 anymore, right?

**Comment by hasancelik** [comment]
> @hasancelik @mesutcelik , so what plan to not support Helm 2 anymore, right?

@leszko yes, that is our plan.

**Comment by leszko** [comment]
@hasancelik it sounds good to me. Is this what other charts do? And what is the deprecation timeline for Helm 2 in general?

**Comment by mesutcelik** [comment]
https://helm.sh/blog/2019-10-22-helm-2150-released/#helm-2-support-plan
https://github.com/helm/charts#deprecation-timeline

Nov 13, 2020 is the end of helm 2. It is already in bugfix mode no new features are added to it. All go to helm 3.

**Comment by leszko** [comment]
Then, it sounds good, let's just remove it.

File: stable/hazelcast-enterprise/Chart.yaml [hunk 1]
```diff
@@ -1,8 +1,7 @@
-apiVersion: v1
+apiVersion: v2
```
[the remark this answers is not in the document]
Comment (hasancelik) [reply stable/hazelcast-enterprise/Chart.yaml:1]: Actually we do not have to use `v2` until we use helm 3 specific resource/method at the charts like:
https://github.com/hazelcast/charts/pull/126#discussion_r425622159

On the other hand, helm 2 is almost dead so we do not need to support it, I think.

[the remark this answers is not in the document]
Comment (leszko) [reply stable/hazelcast-enterprise/Chart.yaml:1]: I checked some random 10 charts on Helm Hub (Redis, Mongo, Jenkins, etc.) and they all are `v1`, which (probably) means that they support Helm 2.  In that case, I'd suggest to just remove Helm 2 mentioned in README, but keep `v1` for some time.

[the remark this answers is not in the document]
Comment (hasancelik) [reply stable/hazelcast-enterprise/Chart.yaml]: I have reverted `apiVersion` to v1 for all charts 👍

**Review by leszko — APPROVED** [review]

**Pull request merged**